airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+    if int(value) <= 0:
+        raise FieldValidationException("The available memory has to be greater than 0")
+
+
+def _validate_max_instances(value):
+    if int(value) <= 0:
+        raise FieldValidationException(
+            "The max instances parameter has to be greater than 0")
+
+
+LABEL_KEY_REGEXP = r'^[\w_-]{0,62}$'
+LABEL_VALUE_REGEXP = r'^[\w_-]{0,63}$'
+
+
+def _validate_labels_dict(the_dictionary):
+    # type: (dict) -> None
+    if not isinstance(the_dictionary, dict):
+        raise FieldValidationException("The field should be dictionary!")
+    for key in the_dictionary.keys():
+        if not re.match(LABEL_KEY_REGEXP, key):
+            raise FieldValidationException("The key '{}' does not match regexp '{}'".
+                                           format(key, LABEL_KEY_REGEXP))
+        value = the_dictionary[key]
+        if not re.match(LABEL_VALUE_REGEXP, value):
+            raise FieldValidationException(
+                "The value '{}' of key '{}' does not match regexp '{}'".
+                format(value, key, LABEL_VALUE_REGEXP))
+
+
+ENV_VARS_KEY_REGEXP = r'^[^\x00=]+$'
+ENV_VARS_VALUE_REGEXP = r'^[^\\x00]*$'
+
+
+def _validate_env_vars_dict(the_dictionary):
+    # type: (dict) -> None
+    if not isinstance(the_dictionary, dict):
+        raise FieldValidationException("The field should be dictionary!")
+    for key in the_dictionary.keys():
+        if key is None:
+            raise FieldValidationException("The key '{}' cannot be None".format(key))
+        if not re.match(ENV_VARS_KEY_REGEXP, key):
+            raise FieldValidationException("The key '{}' does not match regexp '{}'".
+                                           format(key, ENV_VARS_KEY_REGEXP))
+        value = the_dictionary[key]
+        if not re.match(ENV_VARS_VALUE_REGEXP, value):
+            raise FieldValidationException(
+                "The value '{}' of key '{}' does not match regexp '{}'".
+                format(value, key, ENV_VARS_VALUE_REGEXP))
+
+
+CLOUD_FUNCTION_VALIDATION = [
+    dict(name="name", regexp=r'^[^\s]+$'),
+    dict(name="description", regexp="^.+$", optional=True),
+    dict(name="entryPoint", optional=True, regexp=r'^\w+$'),
+    dict(name="runtime", regexp=r'^nodejs6$|^nodejs8$|^python37$', optional=True),
+    dict(name="timeout", regexp=r'^\d+(\.\d{1,9})?s$', optional=True),
+    dict(name="availableMemoryMb", custom_validation=_validate_available_memory_in_mb,
+         optional=True),
+    dict(name="labels", custom_validation=_validate_labels_dict, optional=True),
+    dict(name="environmentVariables", custom_validation=_validate_env_vars_dict,
+         optional=True),
+    dict(name="network",
+         regexp=r'^[^\s]+$',
+         optional=True),
+    dict(name="maxInstances", optional=True, custom_validation=_validate_max_instances),
+
+    dict(name="source_code", type="union", fields=[
+        dict(name="sourceArchiveUrl", regexp=r'^gs://[^\s]+$'),
```

Comment:
This also assumes url has to be gs://, but if later GCF decides to integrate better with, say, github. This may not work but not considered to be a breaking change from GCF side.